src/polygon/src/polygon.cpp

```diff
@@ -747,8 +758,13 @@ class Monotones {
     }
     if (isHole) return true;
 
-    activePairs_.splice(activePairs_.end(), activePairs_, inputPair);
-    inputPair->eastCertain = true;
+    if (activePairs_.size() > 1) {
+      activePairs_.splice(activePairs_.end(), activePairs_, inputPair);
+      inputPair->eastCertain =
+          std::prev(inputPair)->WestOf(vert, precision_) > 0;
```

Comment:
This makes the uncertainty more accurate. `Polygon.Small` still fails for the duplicated case due to an edge being incorrectly marked as certain, but I haven't figured out how to deal with that yet, and it's a pretty nasty test case.